packages/nimble-components/src/table/specs/table-spec.md

```diff
@@ -0,0 +1,242 @@
+
+We will be using TanStack Table to manage all of the table state related to data, sorting, grouping, hierarchy, filtering, and selection. The TanStack APIs provide callback hooks for when certain aspects of the state are changed (e.g. `onSortingChange`), which allows us to update the UI as needed. A typical lifecycle of a user enacted table state change would be, for example:
+
+-   User clicks header to change sort state
+-   Nimble table callback updates sort state in TanStack Table
+-   TanStack Table issues callback that sort state has changed
+-   Nimble table handles callback and retrieves needed rows from TanStack Table model, and updates the UI
+
+### Hooking into TanStack Virtual
+
+TanStack Virtual provides various pieces of state to enable simple, efficient virtualization of table data. The Nimble Table will provide certain state/objects to the TanStack Virtual API for it to then provide the needed state that we can virtualize the table rows with. Namely:
+
+-   The element that will serve as the scollable element
+-   An estimated height for each row
+-   The total count of rows in the data
+
+With this set of information, the Nimble Table will be able to register a callback to the TanStack Virtual `onChange` which will happen any time the scrollable element scrolls. In that handler the Nimble Table can retrieve the set of virtual items from TanStack Virtual (i.e. `getVirtualItems()`), which represent the total set of rows that should be displayed, and contain the state information that allows the Nimble Table to retrieve the appropriate data from the TanStack Table model to apply to each rendered row, as well as the position each row should be rendered.
+
+_Placeholder for other implementation details_
+
+### Accessibility
+
+The `nimble-table` should align to the grid interaction model provided by the [W3C](https://w3c.github.io/aria-practices/#grid).
+
+### Globalization
+
+The order of table columns is inverted in RTL layouts.
+
+### Performance
+
+We intend for the `nimble-table` to handle both the rendering and interactive operations, such as sorting, in a near-instantaneous fashion on datasets of at least 10K rows. Between 10K and 100K rows of data on the client-side, however, we expect to see a notable drop in performance.
+
+Clients, at least initially, will be expected to create mechanisms to restrict the amount of data coming to the client-side.
+
+There are a couple of mechanisms we will leverage to ensure we achieve the necessary performance goals:
+
+-   Usage of the FAST `repeat` directive (avoiding using `positioning: true`)
+-   The [TanStack Virtual](https://github.com/TanStack/virtual) library gives us a simple means of providing the proper state to render the right rows in response to scroll events.
+
+### Dependencies
+
+-   [TanStack Table](https://github.com/tanstack/table)
+-   [TanStack Virtual](https://github.com/TanStack/virtual)
+
+### Test Plan
+
+Intend to test completely with unit tests.
```

Comment:
This section should probably also mention chromatic testing as well.

Reply:
Done.